FWCore/Framework/interface/PathsAndConsumesOfModules.h

```diff
@@ -54,10 +64,20 @@ namespace edm {
     std::vector<ModuleDescription const*> const& doModulesWhoseProductsAreConsumedBy(
         unsigned int moduleID, BranchType branchType) const override;
 
+    std::vector<eventsetup::ComponentDescription const*> const& doESModulesWhoseProductsAreConsumedBy(
+        unsigned int moduleID, Transition) const override;
```

Comment:
Should we change `doModulesWhoseProductsAreConsumedBy` so it also uses `Transition` instead of `BranchType`?

Reply:
The consumes calls for ED products use BranchType. The esConsumes calls for EventSetup products use Transition. I think it has to be this way.

Reply:
Note that Transition would give information on when a product is consumes. BranchType tells whether it is coming from a Run, Lumi or Event. Given that a module can consume a product from a Run during beginLumi, we might need both BranchType and Transition in the consumes call if we want to specify the Transition.

Reply:
We have an open issue about adding transition information in consumes call https://github.com/cms-sw/cmssw/issues/36630 . I'd suggest to leave that development for future, and move further discussion on the topic to the issue.